registry/api/v2/urls.go

```diff
@@ -157,6 +157,18 @@ func (ub *URLBuilder) BuildBlobURL(ref reference.Canonical) (string, error) {
 	return layerURL.String(), nil
 }
 
+// BuildBlobRelativeURL constructs the relative url for the blob identified by name and dgst.
+func (ub *URLBuilder) BuildBlobRelativeURL(ref reference.Canonical) (string, error) {
```

Comment:
I'm not sure this should be in the API for `URLBuilder`. It would make more sense to do something like this:

```
u, err := ub.BuildBlobURL(ref)
if relativeEnabled {
     // make relative
}
```

That, or just configure the `URLBuilder` always return relative urls, which is its natural state.